src/common/statistics/StatisticsManager.ts

```diff
@@ -0,0 +1,394 @@
+        lastRegisteredUrl,
+        lastRegisteredBaseFee,
+        lastRegisteredPctFee,
+        registeredWorkers
+      }
+    }
+    return {
+      ownerBalance,
+      managerBalance,
+      stakeStatus,
+      isRegistered,
+      authorizedHubs,
+      managerAddress,
+      stakeInfo,
+      relayHubEarningsBalance,
+      relayHubEvents,
+      registrationInfo,
+      stakeManagerEvents
+    }
+  }
+
+  getStakeManagerEvents (managerAddress?: Address): StakeMangerEvents {
+    const stakeAddedEvents =
+      this.allStakeManagerEvents.stakeAddedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const stakeUnlockedEvents =
+      this.allStakeManagerEvents.stakeUnlockedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const stakeWithdrawnEvents =
+      this.allStakeManagerEvents.stakeWithdrawnEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const stakePenalizedEvents =
+      this.allStakeManagerEvents.stakePenalizedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const hubAuthorizedEvents =
+      this.allStakeManagerEvents.hubAuthorizedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const hubUnauthorizedEvents =
+      this.allStakeManagerEvents.hubUnauthorizedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const allEvents =
+      this.allStakeManagerEvents.allEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    return {
+      allEvents,
+      stakeAddedEvents,
+      stakeUnlockedEvents,
+      stakeWithdrawnEvents,
+      stakePenalizedEvents,
+      hubAuthorizedEvents,
+      hubUnauthorizedEvents
+    }
+  }
+
+  async fetchStakeManagerEvents (): Promise<void> {
+    const allEvents = await this.contractInteractor.getPastEventsForStakeManager(allStakeManagerEvents, [], { fromBlock: 1 })
+
+    const stakeAddedEvents = this.extractTransactionInfos<StakeAddedEventInfo>(allEvents, StakeAdded)
+    const stakeUnlockedEvents = this.extractTransactionInfos<StakeUnlockedEventInfo>(allEvents, StakeUnlocked)
+    const stakeWithdrawnEvents = this.extractTransactionInfos<StakeWithdrawnEventInfo>(allEvents, StakeWithdrawn)
+    const stakePenalizedEvents = this.extractTransactionInfos<StakePenalizedEventInfo>(allEvents, StakePenalized)
+    const hubAuthorizedEvents = this.extractTransactionInfos<HubAuthorizedEventInfo>(allEvents, HubAuthorized)
+    const hubUnauthorizedEvents = this.extractTransactionInfos<HubUnauthorizedEventInfo>(allEvents, HubUnauthorized)
+
+    this.allStakeManagerEvents = {
+      allEvents,
+      stakeAddedEvents,
+      stakeUnlockedEvents,
+      stakeWithdrawnEvents,
+      stakePenalizedEvents,
+      hubAuthorizedEvents,
+      hubUnauthorizedEvents
+    }
+  }
+
+  async attemptPing (url: string): Promise<PingResult> {
+    let relayPing: PingResult
+    try {
+      const pingResponse = await this.httpClient.getPingResponse(url)
```

Comment:
must be a way to handle timeouts (yes, not here, at the API level of this class)

Reply:
What do you mean? errors are caught in this code, do you mean that httpClient should handle timeouts inside `getPingResponse` function? If so, it shouldn't be part of this PR.

Reply:
It waits for all relayers, meaning a dead relayer can make the API dead for a long time. As a user (eg admin that runs the cli) I need to be able to select hours much time to wait. Right now, there's no such parameter